cmd/snap-confine/snap-confine.c

```diff
@@ -180,6 +180,7 @@ static void sc_restore_process_state(const sc_preserved_process_state *
 	/* If the original working directory cannot be used for whatever reason then
 	 * move the process to a special void directory. */
 	const char *sc_void_dir = "/var/lib/snapd/void";
+	int void_dir_fd SC_CLEANUP(sc_cleanup_close) = -1;
```

Comment:
This is fine, but perhaps it makes sense to declare this nearer its first use. Eg:
```
  the_void:
	/* The void directory may be absent. On core18 system, and other
	 * systems using bootable base snap coupled with snapd snap, the
	 * /var/lib/snapd directory structure is not provided with packages but
	 * created on demand. */
        int void_dir_fd SC_CLEANUP(sc_cleanup_close) = -1;
	void_dir_fd = open(sc_void_dir, O_DIRECTORY | O_PATH | O_NOFOLLOW | O_CLOEXEC);
...
```

Not a blocker.

Reply:
This is invalid C. Labels cannot denote variable declarations. I tried that earlier :)

Reply:
Oh, heh